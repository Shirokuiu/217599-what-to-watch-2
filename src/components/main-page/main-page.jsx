import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import GenreCatalog from "../genre-catalog/genre-catalog";
import MoviePromo from "../movie-promo/movie-promo";
import {Footer} from "../footer/footer";

import {Operation} from "../../reducer";

import {withGenreCatalog} from "../../hocs/with-genre-catalog/with-genre-catalog";
import {withGetStore} from "../../hocs/with-get-store/with-get-store";

const GenreCatalogWrapped = withGenreCatalog(GenreCatalog);
const MoviePromoWrapped = withGetStore(MoviePromo, `promo`);

export const MainPage = (props) => {
  const {history, onLoadPromo} = props;

  onLoadPromo();

  return <React.Fragment>
    <MoviePromoWrapped history={history} />

    <div className="page-content">
      <GenreCatalogWrapped />

      <Footer />
    </div>
  </React.Fragment>;
};

MainPage.propTypes = {
  history: PropTypes.object.isRequired,
  onLoadPromo: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onLoadPromo: () => {
    dispatch(Operation.loadPromo());
  }
});

export default connect(null, mapDispatchToProps)(MainPage);
