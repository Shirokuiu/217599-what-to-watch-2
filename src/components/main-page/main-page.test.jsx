import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {MainPage} from "./main-page";

Enzyme.configure({adapter: new Adapter()});

it(`Is MainPage rendered`, () => {
  const tree = shallow(<MainPage
    history={{}}
    onLoadPromo={jest.fn()}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
