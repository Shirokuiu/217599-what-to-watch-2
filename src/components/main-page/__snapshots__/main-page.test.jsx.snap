// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is MainPage rendered 1`] = `
<Fragment>
  <Connect(WithGetStore)
    history={Object {}}
  />
  <div
    className="page-content"
  >
    <WithGenreCatalog />
    <Footer />
  </div>
</Fragment>
`;
